homeassistant/components/melnor/switch.py

```diff
@@ -21,52 +43,64 @@ async def async_setup_entry(
     async_add_devices: AddEntitiesCallback,
 ) -> None:
     """Set up the switch platform."""
-    switches = []
+    entities: list[MelnorZoneSwitch] = []
 
     coordinator: MelnorDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
 
     # This device may not have 4 valves total, but the library will only expose the right number of valves
     for i in range(1, 5):
-        if coordinator.data[f"zone{i}"] is not None:
-            switches.append(MelnorSwitch(coordinator, i))
-
-    async_add_devices(switches)
-
-
-class MelnorSwitch(MelnorBluetoothBaseEntity, SwitchEntity):
+        valve = coordinator.data[f"zone{i}"]
+        if valve is not None:
+
+            entities.append(
+                MelnorZoneSwitch(
+                    coordinator,
+                    valve,
+                    MelnorSwitchEntityDescription(
+                        device_class=SwitchDeviceClass.SWITCH,
+                        icon="mdi:sprinkler",
+                        key="manual",
+                        name="Manual",
+                        on_fn=lambda valve: setattr(valve, "is_watering", True),
+                        off_fn=lambda valve: setattr(valve, "is_watering", False),
```

Comment:
```suggestion
                        on_fn=lambda valve: valve.is_watering = True,
                        off_fn=lambda valve: valve.is_watering = False,
```

Is `setattr` required here?

Reply:
The linter was giving me grief when I attempted a direct assignment operator

Reply:
`setattr` is hiding the issue that the linter is complaining about.  It should be fixed instead

Reply:
Correct me if I'm wrong, but an assignment statement isn't valid syntax in a python lambda, is it? My understanding was a lambda could only contain an expression. I'm fairly new to python so I'm happy to be wrong here but the suggestion as written above won't even run.

```
2022-09-05 16:09:24.995 ERROR (MainThread) [homeassistant.loader] Unexpected exception importing platform homeassistant.components.melnor.switch
Traceback (most recent call last):
  File "/Users/justin/projects/core/homeassistant/loader.py", line 681, in get_platform
    cache[full_name] = self._import_platform(platform_name)
  File "/Users/justin/projects/core/homeassistant/loader.py", line 698, in _import_platform
    return importlib.import_module(f"{self.pkg_path}.{platform_name}")
  File "/opt/homebrew/Cellar/python@3.9/3.9.13_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 846, in exec_module
  File "<frozen importlib._bootstrap_external>", line 983, in get_code
  File "<frozen importlib._bootstrap_external>", line 913, in source_to_code
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/justin/projects/core/homeassistant/components/melnor/switch.py", line 64
    on_fn=lambda valve: valve.is_watering = True,
                                          ^
SyntaxError: invalid syntax
2022-09-05 16:09:24.996 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform melnor.switch: Platform not found (Exception importing homeassistant.components.melnor.switch).
```

Reply:
Absolutely. You have to use a real function